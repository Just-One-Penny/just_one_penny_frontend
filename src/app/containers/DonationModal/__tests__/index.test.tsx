import React from 'react';
import { render } from '@testing-library/react';
import { Store } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import { configureAppStore } from 'store/configureStore';
import { DonationModal } from '..';

const renderComponent = (store: Store) =>
  render(
    <Provider store={store}>
      <DonationModal charityId={'testId'} charityName={'testCharity'} />
    </Provider>,
  );

describe('<DonationModal />', () => {
  let store: ReturnType<typeof configureAppStore>;

  beforeEach(() => {
    store = configureAppStore();
  });
  it('should match the snapshot', () => {
    const component = renderComponent(store);
    expect(component.container.firstChild).toMatchSnapshot();
  });
});
