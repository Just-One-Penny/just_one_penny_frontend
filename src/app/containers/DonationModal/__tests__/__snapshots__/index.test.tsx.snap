// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DonationModal /> should match the snapshot 1`] = `
.c3 {
  border: 2px solid;
  border-radius: 4px;
  display: block;
  width: 100%;
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.5;
  -webkit-transition: border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  outline: none;
  height: 2.75rem;
  padding: 0 0.75rem;
}

.c3::-webkit-input-placeholder {
  font-weight: 300;
}

.c3::-moz-placeholder {
  font-weight: 300;
}

.c3:-ms-input-placeholder {
  font-weight: 300;
}

.c3::placeholder {
  font-weight: 300;
}

.c1 {
  text-align: center;
  font: normal normal 900 70px/96px Avenir;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  opacity: 1;
}

.c2 {
  text-align: left;
  font: normal normal normal 16px/22px Avenir;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #333333;
  opacity: 1;
}

.c4 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.25rem 1rem;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4:hover {
  opacity: 0.8;
}

.c4:active {
  opacity: 0.4;
}

.c4 .icon {
  margin-right: 0.25rem;
}

.c5 {
  background-color: #c14200;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.25rem 1rem;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5:hover {
  opacity: 0.8;
}

.c5:active {
  opacity: 0.4;
}

.c5 .icon {
  margin-right: 0.25rem;
}

.c0 {
  top: 50px;
  left: 415px;
  width: 450px;
  height: 700px;
  background: #ffffff 0% 0% no-repeat padding-box;
  box-shadow: 0px 3px 6px #00000029;
  border-radius: 5px;
  opacity: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0"
>
  <h2
    class="c1"
  >
    Donate to 
    testCharity
  </h2>
  <p
    class="c2"
  >
    Sunt ad commodo ex magna laboris id sint exercitation cupidatat. Enim anim dolore nostrud ad nulla incididunt magna qui do qui proident cupidatat quis. Sit ad tempor veniam veniam cupidatat anim ullamco voluptate commodo exercitation. Veniam culpa occaecat nulla dolore mollit nostrud aliquip ullamco est ex deserunt. Proident labore est dolor consectetur minim anim incididunt eu ad mollit ipsum elit.
  </p>
  $
  <input
    class="c3"
    min="0.01"
    name="amount"
    placeholder="0.01"
    required=""
    step="0.01"
    type="number"
  />
  <button
    class="c4"
  >
    Donate
  </button>
  <button
    class="c5"
  >
    Cancel
  </button>
</div>
`;
